datalad/core/distributed/clone.py

```diff
@@ -585,15 +585,125 @@ def postclonecfg_ria(ds, props):
+
+        config_content = None
+        scheme = props['source'].split(':')[0][4:]
+        if scheme == 'http':
+
+            # Note: Following outcommented code was proof how to check via HTTP.
+            #       However, for now we only want to annexenable origin in case
+            #       of HTTP, not the special remote. Hence it's uuid isn't
+            #       actually required. (see gh-4410)
+
+            # import requests
+            # try:
+            #     response = requests.get("{}{}config".format(
+            #         props['giturl'],
+            #         '/' if not props['giturl'].endswith('/') else '')
+            #     )
+            #     config_content = response.text
+            # except requests.RequestException as e:
+            #     lgr.debug("Failed to get config file from source:\n%s",
+            #               exc_str(e))
+
+            # config_content still None should lead to skip everything special
+            # remote related. Instead enable origin to deliver annex keys via
+            # HTTP:
+            lgr.debug("Trying to enable remote origin for git-annex")
+            ds.repo.enable_remote('origin')
+            ds.repo.fsck(remote='origin', fast=True)
+
+        elif scheme == 'ssh':
+
+            # TODO: switch the following to proper command abstraction:
+            from datalad.distributed.ora_remote import SSHRemoteIO, RIARemoteError
+            # SSHRemoteIO ignores the path part ATM. No remote CWD! (To be
+            # changed with command abstractions). So we need to get that part to
+            # have a valid path to origin's config file:
+            from datalad.support.network import URL
+            from datalad.utils import PurePosixPath
+            cfg_path = PurePosixPath(URL(props['giturl']).path) / 'config'
+            op = SSHRemoteIO(props['giturl'])
+            try:
+                # TODO: double-check path
+                response = op.read_file(cfg_path)
+                config_content = response
+            except RIARemoteError as e:
+                lgr.debug("Failed to get config file from source:\n%s",
+                          exc_str(e))
+
+        elif scheme == 'file':
+
+            # TODO: switch the following to proper command abstraction:
+            from datalad.distributed.ora_remote import LocalIO, RIARemoteError
+            op = LocalIO()
+            from datalad.support.network import URL
+            cfg_path = Path(URL(props['giturl']).localpath) / 'config'
+            try:
+                # TODO: double-check path
+                response = op.read_file(cfg_path)
+                config_content = response
+            except RIARemoteError as e:
+                lgr.debug("Failed to get config file from source:\n%s",
+                          exc_str(e))
+        else:
+            lgr.debug("Unknown URL-Scheme in %s. Can handle HTTP, SSH or FILE."
+                      "", props['source'])
```

Comment:
What is the `""` for?

Reply:
hehe. Prob. used a `format` call before. In that case I often use `"".format(...` in a new line.